src/nisarqa/__main__.py

```diff
@@ -173,6 +173,71 @@ def load_user_runconfig(runconfig_yaml):
     return user_rncfg
 
 
+def generate_root_params(
```

Comment:
Currently this function (and [`build_root_params()`](https://github.com/isce-framework/nisarqa/blob/550a3c134baec89e4cba42bfcb8883a35a66c8da/src/nisarqa/parameters/rslc_caltools_params.py#L1199)) are free functions.

As an alternative, consider implementing these as class methods of the `nisarqa.RootParamGroup` class, with names such as `RootParamGroup.from_runconfig_dict()` and `RootParamGroup.from_runconfig_file()`.

(I think this should work fine without any circular dependencies, but would need to give it a try to double-check that that's the case.)